lms/djangoapps/courseware/migrations/0008_auto_20190321_1525.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.20 on 2019-03-21 15:25
+from __future__ import unicode_literals
+
+import json
+from django.db import migrations
+
+
+def move_overrides_to_edx_when(apps, schema_editor):
+    from xmodule.fields import Date
+    from edx_when import api
+    date_field = Date()
+    StudentFieldOverride = apps.get_model('courseware', 'StudentFieldOverride')
+    for override in StudentFieldOverride.objects.filter(field='due'):
+        abs_date = date_field.from_json(json.loads(override.value))
+        api.set_date_for_block(
+            override.course_id,
+            override.location,
+            'due',
+            abs_date,
+            user=override.student)
```

Comment:
Please add error catching and debug information here so that if it blows up during a migration, we can know where it failed and on what.

Reply:
I renamed the migration and added exception catching. There's no data to migrate on stage, edge, or prod. So this would only apply to any Open edX installations that use IDDE.